src/AMDGPUOffload.cpp

```diff
@@ -213,19 +212,20 @@ class AMDGPULinker : public Linker {
     }
 
     Symbol add_plt_entry(const Symbol &sym, Section &plt, Section &got, const Symbol &got_sym) override {
-        internal_error << "Unsupported plt relocation for" << sym << "\n";
+        internal_error << "Unsupported plt relocation for amdgpu object" << "\n";
         return sym;
     }
 
     Relocation relocate(uint64_t fixup_offset, char *fixup_addr, uint64_t type,
                         const Elf::Symbol *sym, uint64_t sym_offset, int64_t addend,
-                        Elf::Section &got) override {
-        if (type == R_HEX_32) {
-            // Don't do this relocation, generate a new R_HEX_RELATIVE relocation instead.
-            return Relocation(R_HEX_RELATIVE, fixup_offset, sym_offset + addend, nullptr);
+                        Elf::Section &got, uint64_t got_offset) override {
+        do_relocation(fixup_offset, fixup_addr, type, sym, sym_offset, addend, got, got_offset);
+
+        if (type == R_AMDGPU_ABS64 && sym->is_defined()) {
+            return Relocation(R_AMDGPU_RELATIVE64, fixup_offset, sym_offset + addend, nullptr);
+        } else if (type == R_AMDGPU_ABS32_LO || type == R_AMDGPU_ABS32_HI || type == R_AMDGPU_ABS32 || type == R_AMDGPU_ABS64) {
+            return Relocation(type, fixup_offset, addend, sym);
```

Comment:
@t-tye what do if both the conditionals are false?

Reply:
You return the default constructed Relocation as before. This tells the caller that the relocation record is a static relocation that has been fully processed and will not be included in the linked resulting code object.